test/test_input_recipe.py

```diff
@@ -149,3 +149,29 @@ def testNestedFwd(self):
         self.cmpEntry(res[3], "d", fwd=True)
         self.cmpEntry(res[4], "e")
 
+    def testConditions(self):
```

Comment:
That unit test looks odd. If at all it should be in a separate test suite and in a file called `test_input_env.py`. All these cases simply test the `funMatch` function that is already tested in `test_input_stringparser.py`. Instead a dedicated test for bob.input.Env.evaluate should be created without relying on a particular string function.